src/Microsoft.OData.Core/UriParser/Resolver/NormalizedModelElementsCache.cs

```diff
@@ -89,26 +93,76 @@ public List<IEdmTerm> FindTerms(string qualifiedName)
             return null;
         }
 
+        /// <summary>
+        /// Find all navigation sources that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching navigation sources, or null if no navigation source matches the name.</returns>
+        public List<IEdmNavigationSource> FindNavigationSources(string name)
+        {
+            if (navigationSourcesCache.TryGetValue(name, out List<IEdmNavigationSource> results))
+            {
+                return results;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Find all operation imports that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching operation imports, or null if no operation import matches the name.</returns>
+        public List<IEdmOperationImport> FindOperationImports(string name)
+        {
+            if (operationImportsCache.TryGetValue(name, out List<IEdmOperationImport> results))
+            {
+                return results;
+            }
+
+            return null;
+        }
+
         private void PopulateSchemaElements(IEdmModel model)
         {
             foreach (IEdmSchemaElement element in model.SchemaElements)
             {
                 if (element is IEdmSchemaType schemaType)
                 {
-                    AddElementToCache(schemaType, schemaTypesCache);
+                    AddSchemaElementToCache(schemaType, schemaTypesCache);
                 }
                 else if (element is IEdmOperation operation)
                 {
-                    AddElementToCache(operation, operationsCache);
+                    AddSchemaElementToCache(operation, operationsCache);
                 }
                 else if (element is IEdmTerm term)
                 {
-                    AddElementToCache(term, termsCache);
+                    AddSchemaElementToCache(term, termsCache);
+                }
+            }
+        }
+
+        private void PopulateContainerElements(IEdmModel model)
+        {
+            if (model.EntityContainer is null)

```

Comment:
can the model be null here?

Reply:
It's the responsibility of the caller to ensure that the model is not null here. Since this is an internal class, I placed a `Debug.Assert(model != null)` in the constructor.